pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -262,6 +275,30 @@ func (c *StaticResourceController) Sync(ctx context.Context, syncContext factory
 		return nil
 	}
 
+	ready, err := c.precondition(ctx)
+	if err != nil {
+		if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    fmt.Sprintf("%sDegraded", c.name),
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "ErrorCheckingPrecondition",
+			Message: err.Error(),
+		})); updateErr != nil {
+			return updateErr
+		}
+		return err
+	}
+	if !ready {
+		if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    fmt.Sprintf("%sDegraded", c.name),
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "PreconditionNotReady",
+			Message: "PreconditionNotReady",
+		})); updateErr != nil {
+			return updateErr
+		}
+		return err
+	}
```

Comment:
Will this degraded condition go away ever? Don't see a false-case.

Reply:
This flips back here: https://github.com/openshift/library-go/blob/master/pkg/operator/staticresourcecontroller/static_resource_controller.go#L298